goal_src/engine/ambient/weather-part.gc

```diff
@@ -464,22 +450,8 @@
     (let ((gp-0 (new 'stack-no-clear 'vector)))
       (sp-kill-particle arg0 arg1)
       (set-vector! gp-0 (-> arg2 x) (-> arg1 user-float) (-> arg2 z) 1.0)
-      (sp-launch-particles-var
-        *sp-particle-system-2d*
-        (-> *part-id-table* 39)
-        gp-0
-        (the-as sparticle-launch-state #f)
-        (the-as sparticle-launch-control #f)
-        1.0
-        )
-      (sp-launch-particles-var
-        *sp-particle-system-3d*
-        (-> *part-id-table* 40)
-        gp-0
-        (the-as sparticle-launch-state #f)
-        (the-as sparticle-launch-control #f)
-        1.0
-        )
+      (launch-particles :rate 1.0 (-> *part-id-table* 39) gp-0)
```

Comment:
Why the `:rate 1.0` here, and on a few of the others?

Reply:
Setting the rate is overriding it so it doesn't get set to 0.4. I don't understand why, but setting the rate to 0.4 on these cause it to crash. It seems like this is the case for any function that is like `check-drop-level` but also a couple of other places. I can't notice any difference visually/functionally and haven't really tried to understand what these are doing

This might be a reason that doing it inside the particle functions isn't possible since it can't just set everything to 0.4